code/obj/submachine/robotics.dm

```diff
@@ -122,20 +122,85 @@
 
 /obj/item/atmosporter
 	name = "Atmospherics Transporter"
-	desc = "Used by Atmospherics Cyborgs for convenient transport of siphons and tanks."
-	icon = 'icons/obj/items/items.dmi'
-	icon_state = "bedbin"
+	desc = "Used by cyborgs for convenient transport of siphons and tanks."
+	icon = 'icons/obj/items/device.dmi'
+	icon_state = "atmosporter"
 	var/capacity = 2
 
 	attack_self(var/mob/user as mob)
 		if (src.contents.len == 0) boutput(user, "<span class='alert'>You have nothing stored!</span>")
 		else
 			var/selection = input("What do you want to drop?", "Atmos Transporter", null, null) as null|anything in src.contents
 			if(!selection) return
-			selection:set_loc(user.loc)
-			selection:contained = 0
+			if (istype(selection, /obj/machinery/fluid_canister))
+				var/obj/machinery/fluid_canister/S = selection
+				S.set_loc(user.loc)
```

Comment:
How does this work if the user is in a locker? Should this be using `get_turf`?

Reply:
It does place the thing inside the locker with how things currently are. A more serious problem is that apparently borgs could lose cans and stuff by dropping them while inside pods all this time.

get_turf works for both of those, but the result looks a bit goofy- stuff just materialising on top of closets/pods. I wonder if that's good or if it shouldn't work when inside either.

Reply:
Could always do a check to see if the cyborg is not in anything (i.e. its `loc` is the same as its `get_turf(user)`), else show a message and fail to eject anything?

Reply:
Done (with everything I think)